crates/miden-lib/asm/note_scripts/MINT.masm

```diff
@@ -0,0 +1,58 @@
+use.miden::active_note
+use.miden::contracts::faucets::network_fungible->faucet
+
+# CONSTANTS
+# =================================================================================================
+
+const.MINT_NOTE_INPUTS_NUMBER=9
+
+# ERRORS
+# =================================================================================================
+const.ERR_MINT_WRONG_NUMBER_OF_INPUTS="MINT script expects exactly 9 note inputs"
+
+#! Network Faucet MINT script: mints assets by calling the network faucet's distribute function.
+#! This note is intended to be executed against a network fungible faucet account.
+#!
+#! Requires that the account exposes:
+#! - distribute procedure (from network_fungible faucet contract).
+#!
+#! Inputs:  [ARGS]
+#! Outputs: []
+#!
+#! Note inputs are assumed to be as follows (in order):
+#! - RECIPIENT: The recipient account ID (4 elements)
+#! - execution_hint: Execution hint for the output note
+#! - note_type: Type of the output note
+#! - aux: Auxiliary data for the output note
+#! - tag: Note tag for the output note
+#! - amount: The amount to mint
+#!
+#! Panics if:
+#! - account does not expose distribute procedure.
+#! - the number of inputs is not exactly 9.
+begin
+    dropw
+    # => []
```

Comment:
Related to one of the previous comments: the state before this instruction should be `[ARGS, pad(12)]` and the state after should be `[pad(16)]`. And I would propagate this in the subsequent comments.